### DISCUSSION TITLE
Why Mingw32 is not recognized automatically when MSVC is not found and the variable CC / CCX are set?

### DISCUSSION THREAD

**Discussion by mattia-donato** [pypa/setuptools#4104] [General]
I am trying to install the pypi package "ultradict". I am using `pip install ultradict` in a venv in Windows OS. I have installed mingw32 and I set environments variable it in the PATH and in CC / CXX. I do not understand why I get the error Microsoft Visual Studio 14 not found. Instead if manually I use `python setup.py build_ext --inplace  -compiler=mingw32 -DMS_WIN64` the compilation is possible. Where is the problem?

**Comment by abravalheri** [comment]
Hi @mattia-donato, please note that the compilation in MinGW is not fully flashed out yet.
You can have a look on https://github.com/pypa/distutils/pull/184 for more info. There might be other open issues in the `pypa/distutils` repository as well.
